e2e/e2etests/test_solana_deposit_refund.go

```diff
@@ -0,0 +1,54 @@
+package e2etests
+
+import (
+	"math/big"
+
+	"github.com/gagliardetto/solana-go"
+	"github.com/stretchr/testify/require"
+
+	"github.com/zeta-chain/zetacore/e2e/runner"
+	"github.com/zeta-chain/zetacore/e2e/utils"
+	testcontract "github.com/zeta-chain/zetacore/testutil/contracts"
+	crosschaintypes "github.com/zeta-chain/zetacore/x/crosschain/types"
+)
+
+// TestSolanaDepositAndCallRefund tests deposit of lamports calling a example contract
+func TestSolanaDepositAndCallRefund(r *runner.E2ERunner, args []string) {
+	require.Len(r, args, 1)
+
+	// parse deposit amount (in lamports)
+	// #nosec G115 e2e - always in range
+	depositAmount := big.NewInt(int64(parseInt(r, args[0])))
+
+	// deploy a reverter contract in ZEVM
+	r.Logger.Info("Deploying reverter contract")
```

Comment:
Same let's put a TODO here to remove this in the future

Reply:
agree. We should avoid repeated deployment. Marked https://github.com/zeta-chain/node/pull/2726/commits/92d67f5897576fc119b93838893b0aaf9a43ec08